src/libraries/System.IO.FileSystem/tests/Enumeration/ErrorHandlingTests.cs

```diff
@@ -101,40 +101,40 @@ public void DeleteDirectoryAfterOpening()
             }
         }
 
-    [Fact]
-    [ActiveIssue("https://github.com/dotnet/runtime/issues/40531", TestPlatforms.Browser)]
-    public void VariableLengthFileNames_AllCreatableFilesAreEnumerable()
-    {
-        DirectoryInfo testDirectory = Directory.CreateDirectory(GetTestFilePath());
-        var names = new List<string>();
-
-        for (int length = 1; length < 10_000; length++) // arbitrarily large limit for the test
+        [Fact]
+        public void VariableLengthFileNames_AllCreatableFilesAreEnumerable()
         {
-            string name = new string('a', length);
-            try { File.Create(Path.Join(testDirectory.FullName, name)).Dispose(); }
-            catch { break; }
-            names.Add(name);
-        }
-        Assert.InRange(names.Count, 1, int.MaxValue);
-        Assert.Equal(names.OrderBy(n => n), Directory.GetFiles(testDirectory.FullName).Select(n => Path.GetFileName(n)).OrderBy(n => n));
-    }
+            DirectoryInfo testDirectory = Directory.CreateDirectory(GetTestFilePath());
+            var names = new List<string>();
 
-    [Fact]
-    [ActiveIssue("https://github.com/dotnet/runtime/issues/40531", TestPlatforms.Browser)]
-    public void VariableLengthDirectoryNames_AllCreatableDirectoriesAreEnumerable()
-    {
-        DirectoryInfo testDirectory = Directory.CreateDirectory(GetTestFilePath());
-        var names = new List<string>();
+            var lengthCap = PlatformDetection.IsBrowser ? 256 : 10_000; // On Browser NAME_MAX is 255, otherwise arbitrarily large limit for the test
```

Comment:
So, in memfs, there is no enforcement of `NAME_MAX`, which is 255.  As a result, you'll get 9999 files, 255 of which have increasing "a" names and the rest get their names truncated.  

I'm not sure if this test has much value.

Reply:
As in the test should be removed all together? If its any worth testing enumerability, changing the length upper limit still allows for that.